java/AxivityParser.java

```diff
@@ -0,0 +1,373 @@
+//BSD 2-Clause (c) 2014: A.Doherty (Oxford), D.Jackson, N.Hammerla (Newcastle)
+import java.io.FileInputStream;
+import java.nio.ByteOrder;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.zone.ZoneRules;
+import java.util.concurrent.TimeUnit;
+
+
+public class AxivityParser {
+
+    final private static int EXIT_SUCCESS = 0;
+    final private static int EXIT_FAILURE = 1;
+    final private static boolean USE_PRECISE_TIME = true;
+    final private static int BUFSIZE = 512;
+
+    private String accFile;
+    private String outFile;
+    private String timeZone;
+    private int timeShift;
+
+    private ZonedDateTime sessionStart = null;
+    private boolean isSessionStartDST;
+    private ZoneId zoneId;
+    private ZoneRules rules;
+
+    private int errCounter = 0;
```

Comment:
Is there someway we can return this? Perhaps by returning a particular error value in the npy file so that it can then be captured in the subsequent python processing?

Reply:
Yes, it's part of my TODO list. I was waiting to dig into Actigraph and GENEA to see what other metrics are returned by them and see if I can abstract them.